### PR TITLE
pscanrulesBeta: Source Code Disclosure skip binary'ish responses

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- CHANGELOG > Added change note.
- pscanbea.html > Updated help entry.
- ScourceCodeDisclosureScanRule > Add control characters check in skip conditional.
- ScourceCodeDisclosureScanRuleUnitTest > Add test to assert the new behavior.

## Related Issues
Fixes zaproxy/zaproxy#8191

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/pscanrulesBeta/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## Unreleased
 ### Changed
 - Update minimum ZAP version to 2.14.0.
+- The Source Code Disclosure rule no longer considers response that contain ISO control characters (those which are likely to be binary file types) (Issue 8191).
```
Comment (ricekot) [inline addOns/pscanrulesBeta/CHANGELOG.md:9]:
```suggestion
- The Source Code Disclosure rule no longer considers responses that contain ISO control characters (those which are likely to be binary file types) (Issue 8191).
```

**Comment by kingthorin** [comment]
Tweaked as suggested.

**Review by ricekot — COMMENTED** [review]
How reliable is to assume that a file with ISO control characters is a binary file? Can there be binary files with no ISO control characters? Or text files that include ISO control characters?

**Comment by kingthorin** [comment]
Well 'technically' all files are binary. It's just that some have a limited character set. It's a little harder to give a clear answer these days. Have a look at the summary at the bottom here: http://mdinfotech.net/resources/asciivsbinary.html

Now consider that most text editors do support Unicode these days.

https://en.m.wikipedia.org/wiki/Control_character

https://docs.oracle.com/javase/8/docs/api/java/lang/Character.html#isISOControl-char-

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by ricekot — APPROVED** [review]
Thanks!

**Comment by kingthorin** [comment]
Thanks

**Pull request merged**